pkg/providers/korifi/api/api_test.go

```diff
@@ -0,0 +1,232 @@
+package api
+
+import (
+	"encoding/json"
+	"net/http"
+	"net/http/httptest"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	korifi "github.com/konveyor/asset-generation/pkg/providers/korifi/models"
+)
+
+var _ = Describe("CFAPIClient", func() {
+	var (
+		server       *httptest.Server
+		client       *KorifiAPIClient
+		mockResponse korifi.ListResponse[korifi.AppResponse]
+	)
+
+	BeforeEach(func() {
+		// Mock server setup
+		mockResponse = korifi.ListResponse[korifi.AppResponse]{
+			Resources: []korifi.AppResponse{
+				{GUID: "app-1", Name: "App One"},
+				{GUID: "app-2", Name: "App Two"},
+			},
+		}
+
+		server = httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			Expect(r.URL.Path).To(Equal("/v3/apps"))
+			w.WriteHeader(http.StatusOK)
+			json.NewEncoder(w).Encode(mockResponse)
+		}))
+
+		client = NewKorifiAPIClient(&http.Client{}, server.URL)
+	})
+
+	AfterEach(func() {
+		server.Close()
+	})
+
+	Describe("ListApps", func() {
+		testSpaceGUID := "example93847523894052745"
+		It("should return the list of apps from the API", func() {
+			result, err := client.ListApps(testSpaceGUID)
+
+			Expect(err).ToNot(HaveOccurred())
+			Expect(result.Resources).To(HaveLen(2))
+			Expect(result.Resources[0].GUID).To(Equal("app-1"))
+			Expect(result.Resources[0].Name).To(Equal("App One"))
+			Expect(result.Resources[1].GUID).To(Equal("app-2"))
+			Expect(result.Resources[1].Name).To(Equal("App Two"))
+		})
+		It("should handle empty response gracefully", func() {
+			server.Config.Handler = http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+				w.WriteHeader(http.StatusOK)
+				json.NewEncoder(w).Encode(korifi.ListResponse[korifi.AppResponse]{})
+			})
+
+			result, err := client.ListApps(testSpaceGUID)
+
+			Expect(err).ToNot(HaveOccurred())
+			Expect(result.Resources).To(BeEmpty())
+		})
+
+		It("should handle non-200 status codes gracefully", func() {
+			server.Config.Handler = http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
```

Comment:
maybe you could consider using their API mock server?
https://github.com/cloudfoundry/go-cfclient/blob/4b783f91d3d5d6dc8d6c4054da3f5a9554a416a0/client/app_features_test.go#L18

Reply:
korifi provider deleted